### PR TITLE
feat(getPublishedCollections): [BACK-1640] update getPublishedCollections db call

### PR DESCRIPTION
**Pull request by Herraj**
## Goal

Update the DB call for getting published collections to include logic for `labels` if provided in the `filters`.

## Todos

- [x] Test locally and in DEV

Thanks to the `updateCollection` mutation and `labels` query, I was able to assign labels to an existing collection in dev on the admin side. And then query it using the public query from this PR.

Tickets:

- [BACK-1640]
- [BACK-1648]

[BACK-1640]: https://getpocket.atlassian.net/browse/BACK-1640?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[BACK-1648]: https://getpocket.atlassian.net/browse/BACK-1648?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/public/resolvers/queries/Collection.integration.ts [hunk 1]
```diff
@@ -480,6 +485,202 @@ describe('public queries: Collection', () => {
...
+
+    it('should get collection with the specified label filter', async () => {
+      const testCollection = await createCollectionHelper(db, {
+        title: 'label-test-collection',
+        author,
+        status: CollectionStatus.PUBLISHED,
+        language: CollectionLanguage.EN,
+      });
+
+      const label = await createLabelHelper(db, 'test-label-one');
```
[the remark this answers is not in the document]
Comment (jpetto) [reply src/public/resolvers/queries/Collection.integration.ts:497]: very weird and there's definitely something going on - BUT, good to not let a test issue hold us up. might be a nice ticket for december to look back into this. 😄

**Review by jpetto — APPROVED** [review]
nice! love the test coverage.

has this gone to dev yet to verify?

File: src/public/resolvers/queries/Collection.integration.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (nina-py) [reply src/public/resolvers/queries/Collection.integration.ts:497]: Was the branch created a while back? I ran into this issue too last week and I updated the `clearDb` helper to wipe data from `CollectionLabel` and `Label` tables, after which I had no issues: https://github.com/Pocket/collection-api/blob/323f37f9f61f91782c2300de938bf0f1d315d8b9/src/test/helpers.ts#L316.

Is it worth trying again after merges from `main` into this branch to see if you can still replicate the issue, @Herraj?

[the remark this answers is not in the document]
Comment (jpetto) [reply src/public/resolvers/queries/Collection.integration.ts:497]: herraj and i paired this morning and checked this very thing - the `clearDb` helper did have the code to clear out `collectionLabel` and `label` tables. 😕

[the remark this answers is not in the document]
Comment (nina-py) [reply src/public/resolvers/queries/Collection.integration.ts:497]: That _is_ bizarre! I can see there are `clearDb` calls in both `beforeAll` and `beforeEach`, I'm not sure what else could be done here other than nuking the entire database and re-creating it from scratch then! :)

[the remark this answers is not in the document]
Comment (nina-py) [reply src/public/resolvers/queries/Collection.integration.ts:497]: I second filing a "chores" ticket for this to look into later, no need for it to be a blocker on this PR

**Review by nina-py — APPROVED** [review]

File: src/public/resolvers/queries/Collection.integration.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jpetto) [reply src/public/resolvers/queries/Collection.integration.ts:497]: it is _so_ bizarre - we spent a good part of our pairing on it and could find no cause. i am now frustrated and invested - i have to stop myself from dropping other work to figure it out. 😄

**Pull request merged**